OrbitGl/App.cpp

```diff
@@ -638,13 +637,40 @@ ErrorMessageOr<void> OrbitApp::OnSaveCapture(const std::string& file_name) {
                                   timers_it_end);
 }
 
-ErrorMessageOr<void> OrbitApp::OnLoadCapture(const std::string& file_name) {
+void OrbitApp::OnLoadCapture(const std::string& file_name) {
+  if (open_capture_callback_) {
+    open_capture_callback_();
+  }
   ClearCapture();
+  string_manager_->Clear();
+  thread_pool_->Schedule([this, file_name]() mutable {
+    capture_loading_cancellation_requested_ = false;
+    ErrorMessageOr<void> result =
+        capture_deserializer::Load(file_name, this, &capture_loading_cancellation_requested_);
+
+    if (result.has_error()) {
+      if (open_capture_failed_callback_) {
+        open_capture_failed_callback_();
+      }
+      SendErrorToUi("Error loading capture",
+                    absl::StrFormat("Could not load capture from \"%s\":\n%s", file_name,
+                                    result.error().message()));
+    }
 
-  OUTCOME_TRY(capture_deserializer::Load(file_name, GCurrentTimeGraph));
+    if (open_capture_finished_callback_) {
+      open_capture_finished_callback_();
+    }
+  });
 
   DoZoom = true;  // TODO: remove global, review logic
-  return outcome::success();
+}
+
+void OrbitApp::OnLoadCatpureCanceled() {
+  capture_loading_cancellation_requested_ = true;
+  if (open_capture_failed_callback_) {
+    open_capture_failed_callback_();
+  }
```

Comment:
I am wondering if you should wait here until capture is stopped. before calling ClearCapture.

Reply:
Right, that would be safer. So you mean by introducing a second atomic<bool> that the deserialiser writes?